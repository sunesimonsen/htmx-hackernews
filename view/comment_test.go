package view

import (
	"errors"
	"testing"

	"github.com/alecthomas/assert/v2"
	"github.com/sunesimonsen/htmx-hackernews/mock"
	"github.com/sunesimonsen/htmx-hackernews/model"
)

func TestCommentView(t *testing.T) {
	t.Run("when the repository returns an error", func(t *testing.T) {
		testerr := errors.New("test")
		repo := mock.CommentRepo{Err: testerr}
		view := CommentView{Repo: repo}
		_, err := view.Data(
			mock.Params{"id": "42"},
			mock.Headers{},
			Options{Layout: "content"},
		)
		assert.Equal(t, err, testerr)
	})

	t.Run("when the view renders succesfully", func(t *testing.T) {
		comment := model.Comment{
			By:      "CapitalistCartr",
			Id:      37176230,
			Parent:  37175721,
			Text:    "The link seems to be Slashdotted into oblivion.",
			Time:    1692369360,
			Answers: 42,
		}
		repo := mock.CommentRepo{Comment: comment}
		view := CommentView{Repo: repo}

		data, err := view.Data(
			mock.Params{"id": "37176230"},
			mock.Headers{"Hx-Request": "true"},
			Options{Layout: "content"},
		)

		assert.NoError(t, err)
		assert.Equal(t, data, ViewData[CommentViewData]{
			Template: "comment.gohtml",
			HashKey:  "answers:42",
			Data: CommentViewData{
				Comment:          comment,
				ShowCommentsLink: true,
			},
		})
	})
}
