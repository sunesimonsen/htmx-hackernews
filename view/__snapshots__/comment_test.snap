
[TestCommentView/when_the_view_renders_succesfully - 1]

  <article class="comment">
      The link seems to be Slashdotted into oblivion.
      <footer>
          <span>CapitalistCartr</span>
          <span data-timestamp>1692369360</span>
          
      </footer>
      
  </article>

---
